src/blssignature.cpp

```diff
@@ -80,6 +80,12 @@ BLSInsecureSignature BLSInsecureSignature::DivideBy(const BLSInsecureSignature&
     return result;
 }
 
+BLSInsecureSignature BLSInsecureSignature::Mul(const relic::bn_t n) const {
```

Comment:
I also didn't implement it this way initially because of the overhead of creating new objects. I suppose, since there is no point compression going on here, that this is fast? Have you run the benchmarks?

Reply:
It should be fast as the compiler does return value optimizations here, meaning that it avoids unnecessary copies as much as possible. I've run the benchmarks and did not notice any negative impacts. It's actually slightly faster, but I only did a few bench runs so that might be variance.